### PR TITLE
build: move several linting rules to biome

### PR DESCRIPTION
**Pull request by anonrig**
This PR moves several eslint rules to Biome.

Before:

```
$ eslint static/app tests/js --ext .js,.jsx,.ts,.tsx
✨  Done in 43.09s.
```

After:

```
$ eslint static/app tests/js --ext .js,.jsx,.ts,.tsx
✨  Done in 39.96s.
```

### CODE REVIEW

**Review by scttcper — APPROVED** [review]

File: biome.json [hunk 1]
```diff
@@ -13,6 +13,14 @@
     "enabled": true,
     "rules": {
       "recommended": false,
+      "a11y": {
+        "noBlankTarget": "error"
+      },
+      "correctness": {
+        "noUnreachable": "error",
+        "useHookAtTopLevel": "error",
```
Comment (scttcper) [inline biome.json:21]: is this the only part of rules of hooks?

Comment (scttcper) [reply biome.json:21]: react-hooks/rules-of-hooks does more than just putting things at the top level

Comment (scttcper) [reply biome.json:21]: oh i see `react-hooks/exhaustive-deps` is separate

Comment (anonrig) [reply biome.json:21]: I didn't include exhaustive deps because there are 5000 issues on the sentry repository. 🫡

**Pull request merged**